### PR TITLE
189 frontend custom keyboard numeric only

### PR DESCRIPTION
**Pull request by taylorfinelli**
# What type of PR is this? (Check all that apply)

- [x] ✨**feature**: Introduces completely new code or new features.
- [ ] 🐛**fix**: Implements changes that fix a bug. Ideally, reference an issue if present.
- [ ] ♻️**refactor**: Includes any code-related change that is neither a fix nor a feature.
- [ ] ✅**build**: Encompasses all changes related to the build of the software, including changes to dependencies or the addition of new ones.
- [ ] ⚡️**test**: Pertains to all changes regarding tests, whether adding new tests or modifying existing ones.
- [ ] 🚰**ci**: Involves all changes related to the configuration of continuous integration, such as GitHub Actions or other CI systems.
- [ ] 📚**docs**: Includes all changes to documentation, such as README files, or any other documentation present in the repository.
- [ ] 🗑️**chore**: Captures all changes to the repository that do not fit into the above categories.

# Description
- New keypad for Marcos' workout tracker
- Includes demo page
- unit tests OTW

# Tests
How was this tested?

- [x] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [x] Manual tests
- [ ] Tests were NOT needed
- [ ] Other (explain below)

# Screenshots
https://github.com/UNLV-CS472-672/2024-S-GROUP3-Barbell/assets/85365584/89724388-a4a0-41d6-a64b-cfb751cd36e9

# Documentation

- [ ] Added to README.me
- [ ] Seperate document
- [x] NO documentation needed

Link to external documentation:

# AI Usage
None

### CODE REVIEW

File: apps/expo/src/app/keypadDemo/index.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+export default function KeypadDemo() {
+  const [number, setNumber] = useState('')
+  const [keypadVisible, setKeypadVisible] = useState(false)
+
+  return (
+    <SafeAreaView style={styles.container}>
+      <View>
+        <TextInput
+          style={{ borderWidth: 1, padding: 10, marginBottom: 20, width: '100%' }}
```
Comment (elliotwesoff) [inline apps/expo/src/app/keypadDemo/index.tsx:16]: NBD, but you could move the style object to a const var up above and then reference it here and in line 25.

File: apps/expo/src/components/ui/keypad/Keypad.test.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    expect(getByTestId('test-.')).toBeTruthy()
+    expect(getByTestId('test-backspace')).toBeTruthy()
+  })
+
+  it('checks each key for functionality', () => {
+    const setNumber = jest.fn()
+    const { getByTestId } = render(
+      <Keypad setNumber={setNumber} number='' keypadVisible={true} setKeypadVisible={jest.fn()} />,
+    )
+
```
Comment (elliotwesoff) [inline apps/expo/src/components/ui/keypad/Keypad.test.tsx:33]: Good tests. I like 🐐

File: apps/expo/src/components/ui/keypad/Keypad.tsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    }
+  }
+
+  const { height } = Dimensions.get('window')
+
+  return !keypadVisible ? (
+    <View testID='invisible-test' />
+  ) : (
+    <View className='flex bg-slate-200 p-1' style={{ height: height * 0.35 }}>
+      <View className='flex-1 flex-row justify-between'>
```
Comment (elliotwesoff) [inline apps/expo/src/components/ui/keypad/Keypad.tsx:50]: I think all of these could be refactored with an array and a `map()` call. Would make this super clean.

**Review by elliotwesoff — CHANGES_REQUESTED** [review]
Requesting changes for the refactor at the bottom. If you don't feel like doing it lmk and I'll approve.

**Comment by elliotwesoff** [comment]
Actually on my end it opens a numpad and regular keyboard when an input is selected. Is this the intended behavior?

File: apps/expo/src/app/keypadDemo/index.tsx (hunk 1 above)
Comment (taylorfinelli) [reply apps/expo/src/app/keypadDemo/index.tsx:16]: This is just a demo and is only used for testing functionality of the keypad before merging. It will be deleted after PR is merged

File: apps/expo/src/components/ui/keypad/Keypad.tsx (hunk 3 above)
Comment (taylorfinelli) [reply apps/expo/src/components/ui/keypad/Keypad.tsx:50]: Yeah i originally had it like this but it made unit testing a pAIN ngl so i just reverted back

**Comment by taylorfinelli** [comment]
> Actually on my end it opens a numpad and regular keyboard when an input is selected. Is this the intended behavior?
> 
> 

@elliotwesoff the use case is going to be implemented differently from this page (see prev comment). Don’t worry too much about the layout of the page, just test the functionality/edge cases of the keypad to make sure people can’t do anything wild. If you want to collapse the OS keyboard you can do CTRL+K I think.

**Comment by taylorfinelli** [comment]
> I have taken a good look at the files and could not find anything wrong. Thank you for taking the time to write plenty of test cases to ensure proper functionality of this custom keypad, I appreciate it! 😃

🇸🇻

**Review by elliotwesoff — APPROVED** [review]

**Review by BitsyBirb — APPROVED** [review]
Ran on my end and everything looks good to me; great job on this feature to everybody involved!
